padloper/structure.py

```diff
@@ -4955,6 +4964,86 @@ def from_id(cls, id: int):
         return _vertex_cache[id]
 
 
+'''Some Convenience Functions
+        to make building up structures easier / more readable
+'''
+
+def _tryadd(self):
+    """
+    Utility function to try adding nodes and simply print 
+    already added messages rather than raising errors.
+
+    Returns either added new object, or gets existing one from db.
+    """
+    try:
+        self.add()
+        print(f'added: {self}')
+        return self
+    except VertexAlreadyAddedError as e:
+        print(e)
+        return self.from_db(self.name)
+
+ComponentType.tryadd = _tryadd
+Component.tryadd = _tryadd
+PropertyType.tryadd = _tryadd
+
+def _tryadd_version(self):
```

Comment:
probably a better way to do this so it's not just a copy of the above fn with one line different, but this was fast/easy for now.